File: backend/src/domain/entities/Feedback.js
// The Feedback entity represents a user's feedback, error report, or improvement idea.
class Feedback {
  constructor({
    id,
    ownerId,
    ownerName,
    category,
    message,
    createdAt,
  }) {
    this.id = id;
    this.ownerId = ownerId;
    this.ownerName = ownerName || 'Unknown User';
    this.category = category; // 'Feedback', 'Error', 'Improvement'
    this.message = message;
    this.createdAt = createdAt || new Date().toISOString();
  }

  // Converts this class instance back into a plain JSON object
  toJSON() {
    return {
      id: this.id,
      ownerId: this.ownerId,
      ownerName: this.ownerName,
      category: this.category,
      message: this.message,
      createdAt: this.createdAt,
    };
  }
}

module.exports = Feedback;

File: frontend/lib/features/account/domain/entities/feedback.dart
class UserFeedback {
  final String id;
  final String ownerId;
  final String ownerName;
  final String category;
  final String message;
  final DateTime createdAt;

  UserFeedback({
    required this.id,
    required this.ownerId,
    required this.ownerName,
    required this.category,
    required this.message,
    required this.createdAt,
  });

  factory UserFeedback.fromJson(Map<String, dynamic> json) {
    return UserFeedback(
      id: json['id'] ?? '',
      ownerId: json['ownerId'] ?? '',
      ownerName: json['ownerName'] ?? '',
      category: json['category'] ?? '',
      message: json['message'] ?? '',
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'category': category,
      'message': message,
    };
  }
}
